tests/common/value_operations.h

```diff
@@ -38,6 +38,32 @@ namespace detail {
 template <typename T, size_t N>
 using ArrayT = T[N];
 
+// helper functions to deal with bool type data
+
+template <typename T, typename U>
+void assign_helper(T& left, const U& right) {
+  left = right;
+}
+
+template <typename T>
+typename std::enable_if_t<
+    std::is_same_v<T, bool> || std::is_same_v<T, std::optional<bool>>, void>
+assign_helper(T& left, const int& right) {
+  left = (right % 2 != 0);
+}
+
+template <typename T, typename U>
+bool are_equal_helper(const T& left, const U& right) {
+  return (left == right);
+}
+
+template <typename T>
+typename std::enable_if_t<
+    std::is_same_v<T, bool> || std::is_same_v<T, std::optional<bool>>, bool>
```

Comment:
Instead of `enable_if` it would probably be better to use `if constexpr` inside the function body if the signature is generally the same.

Reply:
> Instead of `enable_if` it would probably be better to use `if constexpr` inside the function body if the signature is generally the same.

enable_if removed